### PR TITLE
feature(list user profiles): implement listing of users if Authenticated

### PR DESCRIPTION
**Pull request by JoshuaKodhe**
### What does this PR do?

Have an api endpoint to get all the user profiles for an authenticated user

### Description of Task to be completed?
**Default**
`GET api/v1/profiles/`

**Paginated**
`GET http://127.0.0.1:8000/api/v1/profiles/?page=2&page_size=1`

### How should this be manually tested?
**Run the server:**
`python manage.py runserver`

**From postman:**
    1. Create multiple users by sending  `post` requests to the 'api/v1/users/' endpoint with the data body 

```source-json
    { "user" :{
	"username": "joe",
	"email": "joe@email.com",
	"password": "Joe12345"
}
}
```

2. Retrieve all users by sending a `get` request to the  '/api/v1/profiles/' endpoint

### What are the relevant pivotal tracker stories?
[#162949001](https://www.pivotaltracker.com/story/show/162949001)

### Screenshots
**Default get request (with no parameters set)**
screenshot 2019-01-30 at 20 35 10

**get request of the second page and a page_size of one (one user's profile a page)**
screenshot 2019-01-30 at 20 39 06

### CODE REVIEW

File: authors/apps/profiles/views.py [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
 from rest_framework.generics import (
-    RetrieveUpdateAPIView, CreateAPIView, RetrieveAPIView)
+    RetrieveUpdateAPIView, CreateAPIView, RetrieveAPIView, ListAPIView)
 from rest_framework.permissions import (
     IsAuthenticatedOrReadOnly, IsAuthenticated)
 from rest_framework.response import Response
 
+from authors.apps.core.pagination import StandardPagination
 from .exceptions import ProfileDoesNotExist
 from rest_framework.exceptions import ValidationError, NotFound
```
Comment (Hesbon5600) [inline authors/apps/profiles/views.py:7]: rest_framework imports should be above local imports

File: authors/apps/core/pagination.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from rest_framework.pagination import PageNumberPagination
+
+
+class StandardPagination(PageNumberPagination):
+    page_size = 10
+    page_size_query_param = 'page_size'
+    max_page_size = 100
```
Comment (Nyakaru) [inline authors/apps/core/pagination.py]: Add a new line here

**Review by charitymarani — COMMENTED** [review]
Fix the travis build

File: authors/apps/profiles/tests/test_list_profiles.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+from rest_framework.views import status
+from ...articles.tests.base_test import TestBaseCase
+
+
+class TestListUserProfiles(TestBaseCase):
+    def test_authenticated_user_list_profiles(self):
+        self.signup_user()
+        self.signup_user2()
```
Comment (eduhmik) [inline authors/apps/profiles/tests/test_list_profiles.py:8]: Hey @JoshuaKodhe, I just want to understand why you have to sign up two users at this instance

Comment (JoshuaKodhe) [reply authors/apps/profiles/tests/test_list_profiles.py:8]: > Hey @JoshuaKodhe, I just want to understand why you have to sign up two users at this instance

The first user will be the one to log in(authenticated user), the second user is to represent other users on the system. The current logged in user will be excluded from the list(my assumption is that they already know their profiles).  so the second user will be returned in this case.

**Review by Nicanor008 — APPROVED** [review]

**Review by Hesbon5600 — APPROVED** [review]

**Review by kzyangiro — APPROVED** [review]

**Pull request merged**